src/_zkapauthorizer/tests/test_plugin.py

```diff
@@ -264,6 +272,38 @@ def test_returns_hashable(self, configuration):
             ),
         )
 
+    @given(timedeltas(min_value=timedelta(seconds=1)), clocks())
+    def test_metrics_written(self, metrics_interval, clock):
+        """
+        When the configuration tells us where to put a metrics .prom file
+        and an interval how often to do so, test that metrics are actually
+        written there after the configured interval.
+        """
+        metrics_path = self.useFixture(TempDir()).join(u"metrics")
+        configuration = {
+            u"prometheus-metrics-path": metrics_path,
+            u"prometheus-metrics-interval": str(int(metrics_interval.total_seconds())),
+            u"ristretto-issuer-root-url": "foo",
+            u"ristretto-signing-key-path": SIGNING_KEY_PATH.path,
+        }
+        announceable = extract_result(
+            storage_server.get_storage_server(
+                configuration,
+                get_anonymous_storage_server,
+                reactor=clock,
+            )
+        )
+        registry = announceable.storage_server._registry
+
+        g = Gauge("foo", "bar", registry=registry)
+        for i in range(2):
+            g.set(i)
+
+            clock.advance(metrics_interval.total_seconds())
+            self.assertThat(
+                metrics_path, FileContains(matcher=Contains("foo {}".format(i)))
```

Comment:
Consider
```suggestion
                metrics_path, FileContains(matcher=AfterPreprocessing(
                    lambda f: list(text_string_to_metric_families(f)),
                    MatchesListwise([MatchesStructure(name=Equals('foo'), samples=MatchesListwise([MatchesStructure.byEquality(name='foo', value=i)]))]),
                 )
```
where `text_string_to_metric_families` is from `prometheus_client.parser`.
(typed in directly in github, after experimenting in a repl)